This is the second test - a test of proportions to determine whether those in the program group have a higher level of employment in unsubsidized jobs, three years after completing the program.

In [9]:
import numpy as np

alpha=0.05
P_0=0.521
P_1=0.533

n_0=409
n_1=564

In [10]:
sp=(P_0*n_0+P_1*n_1)/(n_1+n_0)
print sp

0.527955806783


In [11]:
sp_stdev= lambda p, n: np.sqrt(p*(1-p)/n)

sp_stdev_3y=sp_stdev(P_0,n_0+n_1)
print P_0, n_0, P_1, n_1, sp_stdev_3y

0.521 409 0.533 564 0.0160151200397


In [12]:
zscore = lambda p0, p1, s : (p0-p1)/s
z_3y = zscore(P_1, P_0, sp_stdev_3y)
print z_3y

0.749291917278


Based on this observed z score, I looked on table and found P = 0.7734 for z score = .75 and alpha = .05

The next step is to find the p value to which this z score correlates, by taking the inverse of the P above.

In [13]:
p_3y=1-.7734
print p_3y

0.2266


In [14]:
def report_result(p,a):
    print 'Is the p value {0:.2f} smaller than the critical value {1:.2f}? '.format(p,a)
    if p<a:
        print "YES!"
    else: print "NO!"
    
    print 'the Null hypothesis is {}'.format( 'rejected' if p<a  else 'not rejected') 

 
report_result(p_3y,alpha)

Is the p value 0.23 smaller than the critical value 0.05? 
NO!
the Null hypothesis is not rejected


This statistical test shows that we cannot reject the null hypothesis, meaning there is not statistically significant evidence of greater employment three years after progam completion.

The next test is a chi-square test to see if employment is higher six quarters later for those who completed the program (the test sample).

|employed in subsidized job |     employed   | not employes   |                   
|---------------------------|----------------|------------------|---------------------------|
| test sample               |  $14.7 * 0.01 * 564 = 82.9$ |   $85.3 * 0.01 * 564 = 481.1$  | 564                       |
| control sample            |  $11.9 * 0.01 * 409 = 48.7$ |   $88.1 * 0.01 * 409 = 360.3$   | 409                       |
|                           |                |                  |                           |
| total                     |    131.6    |     841.4     | 973         |

In [15]:
Ntot = 973
expected = 564*131.6*409*841.4
sample_values = [[14.7*5.64,85.3*5.64],[11.9*4.09,88.1*4.09]]
 
chisqstat= lambda N, values, expect : N*((values[0][0]*values[1][1]-values[0][1]*values[1][0])**2)/(expect)

chisq = chisqstat(Ntot,  sample_values, expected)

print chisq

1.58917936772


critical value for 1 deree of freedom and alpha = .05 is 3.84

In [16]:
chisqcrit=3.84

In [17]:
def report_result(chisq, chisqcrit):
    print 'Is the ChiSq value {0:.2f} larger than the critical value {1:.2f}? '.format(chisq,chisqcrit)
    if chisq>chisqcrit:
        print "YES!"
    else: print "NO!"
    
    print 'the Null hypothesis is {}'.format( 'rejected' if chisq>chisqcrit  else 'not rejected')
    
report_result(chisq, chisqcrit)

Is the ChiSq value 1.59 larger than the critical value 3.84? 
NO!
the Null hypothesis is not rejected


Because the test stastic of ChiSq=1.59 is not larger than the critical value of 3.84 (at p = 0.05), we cannot reject the null hypothesis. This means that we cannot conclude that those completed the program have a higher rate of employment than those who did not complete the program six quarters later.